### PR TITLE
Refactor scripts to use django queries [OSF-8313]

### PR DESCRIPTION
**Pull request by erinspace**
## Purpose
Many of our regularly run scripts are using some modular odm queries - let's make things more consistent

## Changes
- Move some Q queries to django Q queries
- Use django syntax where possible
- Fix unrun scripts tests for sanity checks on regularly run scripts

## Side effects
None anticipated

## Ticket
https://openscience.atlassian.net/browse/OSF-8313

### CODE REVIEW

File: scripts/tests/test_user_summary.py [hunk 1]
```diff
@@ -51,10 +40,13 @@ def setUp(self):
         u.date_disabled = self.a_while_ago
         u.save()
 
-        modify_user_dates_in_mongo(self.yesterday)
+        for user in OSFUser.objects.all():
```
Comment (caseyrollins) [inline scripts/tests/test_user_summary.py]: `OSFUser.objects.all().update(date_registered=self.yesterday)`?

Comment (caseyrollins) [inline scripts/tests/test_user_summary.py]: `OSFUser.objects.all().update(date_registered=self.yesterday)`?

File: scripts/tests/test_user_summary.py [hunk 2]
```diff
@@ -83,7 +75,10 @@ def test_merged_user(self):
         merged_user.save()
         user.reload()
         merged_user.reload()
-        modify_user_dates_in_mongo(self.yesterday)
+
+        for user in OSFUser.objects.all():
```
Comment (caseyrollins) [inline scripts/tests/test_user_summary.py]: `OSFUser.objects.all().update(date_registered=self.yesterday)`?

File: scripts/tests/test_user_summary.py [hunk 3]
```diff
@@ -1,28 +1,17 @@
...
-                {'$set': {
-                    'date_registered': new_date
-                }}
-            )
-
 class TestUserCount(OsfTestCase):
     def setUp(self):
-        self.yesterday = datetime.today() - timedelta(1)
-        self.a_while_ago = datetime.today() - timedelta(2)
+        self.yesterday = timezone.now() - timedelta(1)
```
Comment (caseyrollins) [inline scripts/tests/test_user_summary.py]: Just because I'm curious, why `timezone` over `datetime`?

File: scripts/tests/test_send_queued_mails.py [hunk 4]
```diff
@@ -5,37 +5,38 @@
...
 from scripts.send_queued_mails import main, pop_and_verify_mails_for_each_user, find_queued_mails_ready_to_be_sent
-from website import mails, settings
+from website import settings
 
 class TestSendQueuedMails(OsfTestCase):
 
     def setUp(self):
         super(TestSendQueuedMails, self).setUp()
-        mails.QueuedMail.remove()
+        # QueuedMail.remove()
```
Comment (caseyrollins) [inline scripts/tests/test_send_queued_mails.py]: Should this be uncommented? Or placed in a `tearDown` method?

File: scripts/tests/test_send_queued_mails.py [hunk 5]
```diff
@@ -49,9 +50,14 @@ def test_pop_and_verify_mails_for_each_user(self):
...
         user_with_multiple_emails = UserFactory()
         user_with_no_emails_sent = UserFactory()
-        mail_sent = mails.QueuedMail(user=user_with_email_sent,
-                                     sent_at=timezone.now() - timedelta(days=1),
-                                     to_addr=user_with_email_sent.username)
+        time = timezone.now() - timedelta(days=1)
+        mail_sent = QueuedMail(
+            user=user_with_email_sent,
+            send_at=time,
+            sent_at=time,
```
Comment (caseyrollins) [inline scripts/tests/test_send_queued_mails.py]: Repeat of line above.

File: scripts/analytics/addon_snapshot.py [hunk 6]
```diff
@@ -33,17 +32,17 @@ def get_enabled_authorized_linked(user_settings_list, has_external_account, shor
...
+            password__isnull=False,
+            merged_by__isnull=True,
+            date_disabled__isnull=True,
+            date_confirmed__isnull=False
         ).count()
 
     elif short_name == 'forward':
-        num_enabled = num_authorized = ForwardNodeSettings.find().count()
-        num_linked = ForwardNodeSettings.find(Q('url', 'ne', None)).count()
+        num_enabled = num_authorized = ForwardNodeSettings.objects.all().count()
```
Comment (caseyrollins) [inline scripts/analytics/addon_snapshot.py]: Super minor: this can just be `ForwardNodeSettings.objects.count()`

File: scripts/analytics/institution_summary.py [hunk 7]
```diff
@@ -53,7 +50,7 @@ def get_events(self, date):
                     'name': institution.name,
                 },
                 'users': {
-                    'total': OSFUser.find(user_query).count(),
+                    'total': OSFUser.objects.filter(user_query).count(),
```
Comment (caseyrollins) [inline scripts/analytics/institution_summary.py]: `institution.contributors.count()`?

File: scripts/analytics/institution_summary.py [hunk 8]
```diff
@@ -53,7 +50,7 @@ def get_events(self, date):
...
                 },
                 'nodes': {
                     'total': AbstractNode.find_by_institutions(institution, node_query).count(),
```
Comment (caseyrollins) [inline scripts/analytics/institution_summary.py]: Could all of these use the reverse relation? i.e. `institution.nodes.filter(node_query).count()`?

File: scripts/tests/test_user_summary.py [hunk 9]
```diff
@@ -51,10 +40,13 @@ def setUp(self):
...
 
-        modify_user_dates_in_mongo(self.yesterday)
+        for user in OSFUser.objects.all():
+            user.date_registered = self.yesterday
+            user.save()
+
 
     def tearDown(self):
-        User.remove()
+        OSFUser.objects.all().delete()
```
Comment (sloria) [inline scripts/tests/test_user_summary.py]: Should no longer be necessary, since each test is run in a transaction which is rolled back at the end of the test.

File: scripts/analytics/user_summary.py [hunk 10]
```diff
@@ -18,10 +17,10 @@
 
 # Modified from scripts/analytics/depth_users.py
 def count_user_logs(user):
-    logs = NodeLog.find(Q('user', 'eq', user._id))
+    logs = NodeLog.objects.filter(user=user)
```
Comment (caseyrollins) [inline scripts/analytics/user_summary.py]: `user.logs.all()`?

File: scripts/osfstorage/glacier_audit.py [hunk 11]
```diff
@@ -66,11 +66,10 @@ def get_job(vault, job_id=None):
 
 
 def get_targets(date):
-    return FileVersion.find(
-        Q('date_created', 'lt', date - DELTA_DATE) &
-        Q('status', 'ne', 'cached') &
-        Q('metadata.archive', 'exists', True) &
-        Q('location', 'ne', None)
+    return FileVersion.objects.filter(
```
Comment (caseyrollins) [inline scripts/osfstorage/glacier_audit.py]: I'm not certain, but I think `metadata__archive__isnull=False` should be `metadata__has_key='archive'` instead.

https://docs.djangoproject.com/en/1.11/ref/contrib/postgres/fields/#has-key

File: scripts/send_queued_mails.py [hunk 12]
```diff
@@ -49,18 +48,20 @@ def main(dry_run=True):
 
 
 def find_queued_mails_ready_to_be_sent():
-    return QueuedMail.find(
-        Q('send_at', 'lt', timezone.now()) &
-        Q('sent_at', 'eq', None)
-    )
+    return QueuedMail.objects.filter(send_at__lt=timezone.now(), sent_at__isnull=True)
+
+    # return QueuedMail.find(
```
Comment (caseyrollins) [inline scripts/send_queued_mails.py]: Remove?

File: scripts/send_queued_mails.py [hunk 13]
```diff
@@ -49,18 +48,20 @@ def main(dry_run=True):
...
+    # )
 
 
 def pop_and_verify_mails_for_each_user(user_queue):
     for user_emails in user_queue.values():
         mail = user_emails[0]
-        mails_past_week = QueuedMail.find(
-            Q('user', 'eq', mail.user) &
-            Q('sent_at', 'gt', timezone.now() - settings.WAIT_BETWEEN_MAILS)
+        mails_past_week = QueuedMail.objects.filter(
```
Comment (caseyrollins) [inline scripts/send_queued_mails.py]: `mail.user.queuedmail_set.filter(sent_at__gt=...)`?

File: scripts/tests/test_node_log_events.py [hunk 14]
```diff
@@ -27,7 +29,7 @@ def setUp(self):
 
         self.end_date = datetime.datetime.utcnow() - datetime.timedelta(1)
 
-        for node_log in NodeLog.find():
+        for node_log in NodeLog.objects.all():
```
Comment (caseyrollins) [inline scripts/tests/test_node_log_events.py]: `.update()` here?

File: scripts/tests/test_node_summary.py [hunk 15]
```diff
@@ -56,21 +44,24 @@ def setUp(self):
...
+            node.save()
+        # modify_node_dates_in_mongo(self.date - datetime.timedelta(0.1))
 
         self.results = NodeSummary().get_events(self.date.date())[0]
 
     def tearDown(self):
         super(TestNodeCount, self).tearDown()
-        Node.remove()
-        User.remove()
+        AbstractNode.remove()
```
Comment (caseyrollins) [inline scripts/tests/test_node_summary.py]: Should these be `.remove()` or `.objects.all().delete()`? (I don't know what MODM's `remove()` does)

File: scripts/tests/test_node_summary.py [hunk 16]
```diff
@@ -56,21 +44,24 @@ def setUp(self):
...
+        OSFUser.remove()
 
     def test_get_node_count(self):
         nodes = self.results['nodes']
 
-        assert_equal(nodes['total'], 8)  # 3 Projects, 5 Registrations
+        assert_equal(nodes['total'], 3)  # 2 Projects, 1 component
```
Comment (caseyrollins) [inline scripts/tests/test_node_summary.py]: What change caused `nodes['total']` to change from 8 to 3?

File: scripts/tests/test_user_summary.py (hunk 3 above)
Comment (erinspace) [inline scripts/tests/test_user_summary.py]: datetime wasn't including the timezone information! So I was getting "can't compare naive and aware datetimes" when trying to do date math.

File: scripts/tests/test_node_summary.py (hunk 16 above)
Comment (erinspace) [inline scripts/tests/test_node_summary.py]: This was a mistake! Nodes should never have been counting registrations as well, so the 8 was including registrations. 3 is what the number should have been without registrations.

Comment (caseyrollins) [inline scripts/tests/test_node_summary.py]: Ahhh, I see!

File: scripts/osfstorage/glacier_audit.py (hunk 11 above)
Comment (erinspace) [inline scripts/osfstorage/glacier_audit.py]: done! Good catch!

File: scripts/embargo_registrations.py [hunk 17]
```diff
@@ -24,12 +24,12 @@
...
 
 def main(dry_run=True):
-    pending_embargoes = Embargo.objects.filter(state=Embargo.UNAPPROVED)
+    pending_embargoes = Embargo.objects.select_for_update().filter(state=Embargo.UNAPPROVED)
     for embargo in pending_embargoes:
         if should_be_embargoed(embargo):
             if dry_run:
                 logger.warn('Dry run mode')
-            parent_registration = Registration.objects.get(embargo=embargo)
+            parent_registration = Registration.objects.select_for_update().get(embargo=embargo)
```
Comment (sloria) [inline scripts/embargo_registrations.py]: Drive-by: Are you sure that parent_registration is modified in this script? `select_for_update` should only be used when you are making an update.

File: scripts/analytics/addon_snapshot.py [hunk 18]
```diff
@@ -33,17 +32,17 @@ def get_enabled_authorized_linked(user_settings_list, has_external_account, shor
...
+            merged_by__isnull=True,
+            date_disabled__isnull=True,
+            date_confirmed__isnull=False
         ).count()
 
     elif short_name == 'forward':
-        num_enabled = num_authorized = ForwardNodeSettings.find().count()
-        num_linked = ForwardNodeSettings.find(Q('url', 'ne', None)).count()
+        num_enabled = num_authorized = ForwardNodeSettings.objects.count()
+        num_linked = ForwardNodeSettings.objects.filter(url__isnull=True).count()
```
Comment (sloria) [inline scripts/analytics/addon_snapshot.py]: This should be `url__isnull=False`, I think.

File: scripts/analytics/institution_summary.py [hunk 19]
```diff
@@ -53,27 +50,27 @@ def get_events(self, date):
...
                 },
                 'users': {
-                    'total': OSFUser.find(user_query).count(),
+                    'total': institution.contributors.count(),
                 },
                 'nodes': {
-                    'total': AbstractNode.find_by_institutions(institution, node_query).count(),
-                    'public': AbstractNode.find_by_institutions(institution, node_public_query).count(),
-                    'private': AbstractNode.find_by_institutions(institution, node_private_query).count(),
+                    'total': institution.nodes.filter(node_query).exclude(type='osf.registration').count(),
```
Comment (sloria) [inline scripts/analytics/institution_summary.py]: Good catch!

File: scripts/approve_embargo_terminations.py [hunk 20]
```diff
@@ -31,16 +30,16 @@
...
+        initiation_date__lt=auto_approve_time,
+        state=models.EmbargoTerminationApproval.UNAPPROVED
     )
 
 def main():
     pending_embargo_termination_requests = get_pending_embargo_termination_requests()
     count = 0
     for request in pending_embargo_termination_requests:
-        registration = models.Registration.find_one(Q('embargo_termination_approval', 'eq', request))
+        registration = models.Registration.objects.select_for_update().get(embargo_termination_approval=request)
```
Comment (sloria) [inline scripts/approve_embargo_terminations.py]: Same here: Is `registration` modified? If not, remove select_for_update.

Comment (sloria) [inline scripts/approve_embargo_terminations.py]: Please check this in the other registration scripts as well.

File: scripts/embargo_registrations.py [hunk 21]
```diff
@@ -25,12 +24,12 @@
 
 
 def main(dry_run=True):
-    pending_embargoes = Embargo.find(Q('state', 'eq', Embargo.UNAPPROVED))
+    pending_embargoes = Embargo.objects.filter(state=Embargo.UNAPPROVED)
```
Comment (sloria) [inline scripts/embargo_registrations.py]: I think `select_for_update` *is* appropriate here, since the embargoes are being updated.

File: scripts/embargo_registrations.py [hunk 22]
```diff
@@ -25,12 +24,12 @@
...
 
 def main(dry_run=True):
-    pending_embargoes = Embargo.find(Q('state', 'eq', Embargo.UNAPPROVED))
+    pending_embargoes = Embargo.objects.select_for_update().filter(state=Embargo.UNAPPROVED)
     for embargo in pending_embargoes:
         if should_be_embargoed(embargo):
             if dry_run:
                 logger.warn('Dry run mode')
-            parent_registration = Registration.find_one(Q('embargo', 'eq', embargo))
+            parent_registration = Registration.objects.select_for_update().get(embargo=embargo)
```
Comment (sloria) [inline scripts/embargo_registrations.py]: Should `select_for_update` be used here?

**Comment by sloria** [comment]
Failing tests are unrelated to this PR and have since been fixed on develop. Feel free to update with develop, but it's not strictly necessary. Just one more comment above.
